server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -337,17 +343,23 @@ protected ConstraintSecurityHandler configureSpnego(Server server,
       AvaticaServerConfiguration config) {
     final String realm = Objects.requireNonNull(config.getKerberosRealm());
     final String principal = Objects.requireNonNull(config.getKerberosPrincipal());
-
-    // A customization of SpnegoLoginService to explicitly set the server's principal, otherwise
-    // we would have to require a custom file to set the server's principal.
-    PropertyBasedSpnegoLoginService spnegoLoginService =
-        new PropertyBasedSpnegoLoginService(realm, principal);
+    final String principalWithoutRealm = principal.split("@", 2)[0];
+    final String[] principalParts = principalWithoutRealm.split("/", 2);
+    final String serviceName = principalParts[0];
+    final String hostname = principalParts[1];
+    final Path keytabPath = config.getKerberosKeytab();
+
+    ConfigurableSpnegoLoginService spnegoLoginService =
+        new ConfigurableSpnegoLoginService(realm, new TestAuthorizationService());
+    spnegoLoginService.setServiceName(serviceName);
+    spnegoLoginService.setHostName(hostname);
+    spnegoLoginService.setKeyTabPath(keytabPath);
```

Comment:
It looks like Avatica server currently supports logging in without a keytab. Jetty wants to now manage the Kerberos lifecycle and login.

Reply:
Ugh. That's going to make unit testing harder.